src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py

```diff
@@ -200,11 +200,17 @@
 """
 helps['ad app permission grant'] = """
     type: command
-    short-summary: Grant the app an API permission
+    short-summary: Grant the app an API Delegated permissions
+    long-summary: for Application permissions, please use "ad app permission admin-consent"
     examples:
         - name: Grant a native application with permissions to access an existing API with TTL of 2 years
           text: az ad app permission grant --id e042ec79-34cd-498f-9d9f-1234234 --api a0322f79-57df-498f-9d9f-12678 --expires 2
 """
+helps['ad app permission admin-consent'] = """
+    type: command
+    short-summary: grant Application & Delegated permissions through admin-consent.
+    long-summary: you must login as a directory administrator
```

Comment:
This seems like a good detail to include. When we present help text, do we always present the short summary when we present the long summary? If not, I think we should repeat the short summary there as well.

Reply:
Long summary will always be present from command's help. It will not be visible from a category's help such as `az ad permission -h`.
This should be good enough